charts/aws-adapter/templates/deployment.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "nirmata-aws-adapter.fullname" . }}
+  namespace: {{ .Values.namespace.name }}
+  labels:
+    control-plane: controller-manager
+    app.kubernetes.io/name: deployment
+    app.kubernetes.io/instance: controller-manager
+    app.kubernetes.io/component: manager
+    app.kubernetes.io/created-by: nirmata-aws-adapter
+    app.kubernetes.io/part-of: nirmata-aws-adapter
+spec:
+  selector:
+    matchLabels:
+      control-plane: controller-manager
+  replicas: 1
+  template:
+    metadata:
+      annotations:
+        kubectl.kubernetes.io/default-container: manager
+      labels:
+        control-plane: controller-manager
+    spec:
+      imagePullSecrets:
+      - name: {{ include "nirmata-aws-adapter.fullname" . }}
+      containers:
+      - command:
+        - /manager
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        name: manager
+        {{- if .Values.pollInterval }}
+        args:
+        - --sync-period={{ .Values.pollInterval }}
+        {{- end }}
+        securityContext:
+          runAsNonRoot: true
+          privileged: false
+          allowPrivilegeEscalation: false
+          readOnlyRootFilesystem: true
+          capabilities:
+            drop:
+              - ALL
+          seccompProfile:
+            type: RuntimeDefault
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: 8081
+          initialDelaySeconds: 15
+          periodSeconds: 20
+        readinessProbe:
+          httpGet:
+            path: /readyz
+            port: 8081
+          initialDelaySeconds: 5
+          periodSeconds: 10
+        # TODO(user): Configure the resources accordingly based on the project requirements.
+        # More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+        resources:
+          limits:
+            cpu: 500m
+            memory: 128Mi
+          requests:
+            cpu: 10m
+            memory: 64Mi
+      serviceAccountName: nirmata-aws-adapter-sa
+      terminationGracePeriodSeconds: 10
```

Comment:
Mostly, the deployment will fail some policy checks like the PSS restricted profile. Or even the best practices and other kyverno standard policy sets. Need to add those fields. E.g. see https://github.com/nirmata/imagekey-controller/commit/ac6d87e287e5157bade549b9ecbb4a75cba9306a#diff-ce0d597a9e5f20a8fe61c3ba5e6185a6b90cfef2c00bb5b5ef3e97b15b79f480

Reply:
@pns-nirmata - do you mean these?
https://github.com/nirmata/kyverno-charts/pull/29/files#diff-e2c2d2700e0fb4fdec6d2b4b25e08a4f6956e10d2ae113ad5cb860e6c9df0c20R37-R46

Reply:
Yes. That one. Could be others. Run a `trivy config .` Additionally, just run the pod. Install default policies (best practices, security, multitenancy) in Kyverno and check for other violations.